landingai/pipeline/frameset.py

```diff
@@ -150,6 +157,16 @@ def overlay_predictions(self, options: Optional[Dict[str, Any]] = None) -> "Fram
         )
         return self
 
+    def crop_predictions(self) -> List[Tuple[str, "Frame"]]:
```

Comment:
Why not returning a new FrameSet? I think this would nicely streamline the workflow since the next thing a user would want to do is call FrameSet.predict(ocr...)

Reply:
My first though was using FrameSet for that, but I had 2 concerns:
1. `FrameSet` seems to describe more an arrangement of frames in sequential order (like videos), and less like a `List` (in the pythonic way of saying it)...
2. (more important): if we return a `FrameSet` here, we will lose completly track of which class was associated with that cropped region. That's why this returns a list of tuples with `(label, Frame)`

Reply:
I disagree with #1... a set is unordered :) (a sequence would be ordered)..... #2 worries me more (I did not realize it before). I think we need to try to keep the metadata in the Frame.... why not just copying the prediction to the resulting frame (i.e. a bbox that covers the whole image)?

In general, the more we can keep all the data & metadata mapped to frame & frameset, the more composable the solution will be and less code that the customer needs to add. I would map the predictions to the resulting frames and return a frameset

Reply:
Maybe `FrameSet` class received an unfortunate name, because it has `FrameSet.save_video()`  method. The frames cannot be unordered, right? There is even a cool video resizing example [in the docs](https://landing-ai.github.io/landingai-python/image-acquisition/video-file/), that users can even extend to [adjust color, brightness, etc now](https://github.com/landing-ai/landingai-python/pull/154). :D

I'm not 100% sure about copying also the predictions, tbh. One of the use cases could be running new predictions on image parts, for example. The image parts looks a lot more like a new Frame to me.

Thinking here, maybe we could return a list of `(Prediction, Frame)` (instead of just the prediction text)...

Reply:
Mmm you are right that we are using it to save videos (concept drift 😞 )...  Maybe what we can strive for is that it is easy for the customer to get back to a FrameSet... maybe frameset needs a constructor that takes [(Prediction, Frame)] ....  this way the user can decide to keep or not the predictions and to continue with a framset or just a list